### PR TITLE
[CI][URGENT] Fixes for docker cache generation

### PR DESCRIPTION
**Pull request by larroy**
## Description ##

 * extract arguments from environment
 * control constants via environment
 * add commandline options
 * Fix executable bits of docker/install/ubuntu_publish.sh (fails build publish.ubuntu1404_cpu)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: ci/docker_cache.py [hunk 1]
```diff
@@ -34,11 +32,11 @@
...
-DOCKERHUB_LOGIN_NUM_RETRIES = 5
-DOCKERHUB_RETRY_SECONDS = 5
-DOCKER_CACHE_NUM_RETRIES = 3
-DOCKER_CACHE_TIMEOUT_MINS = 15
-PARALLEL_BUILDS = 10
+DOCKERHUB_LOGIN_NUM_RETRIES =   int(os.environ.get('DOCKER_CACHE_DOCKERHUB_LOGIN_NUM_RETRIES', '5'))
+DOCKERHUB_RETRY_SECONDS =       int(os.environ.get('DOCKER_CACHE_DOCKERHUB_RETRY_SECONDS', '5'))
+DOCKER_CACHE_NUM_RETRIES =      int(os.environ.get('DOCKER_CACHE_DOCKER_CACHE_NUM_RETRIES', '3'))
+DOCKER_CACHE_TIMEOUT_MINS =     int(os.environ.get('DOCKER_CACHE_DOCKER_CACHE_TIMEOUT_MINS', '15'))
+PARALLEL_BUILDS =               int(os.environ.get('DOCKER_CACHE_PARALLEL_BUILDS', '10'))
```
Comment (marcoabreu) [inline ci/docker_cache.py]: Jenkins environment variables are not versioned so far and thus can be changed without notice. For configuration like this, I'd prefer if we would have it in versioned configuration if you don't mind.

File: ci/docker_cache.py [hunk 2]
```diff
@@ -240,16 +234,36 @@ def script_name() -> str:
...
     parser.add_argument("--docker-registry",
                         help="Docker hub registry name",
-                        type=str,
-                        required=True)
+                        default="mxnetci",
+                        type=str)
+
+    parser.add_argument("--docker-secret-name",
+                        help="secrets manager name",
+                        default="prod/DockerCache/DockerHubCredentials",
```
Comment (marcoabreu) [inline ci/docker_cache.py]: The environment variables were chosen because they are environment specific variables (think about prod and test config).

**Comment by marcoabreu** [comment]
Could you separate the urgent fix (changing the executable bit) from the rest of the changes in this PR so we can merge the fix quickly?

**Comment by sandeep-krishnamurthy** [comment]
@marcoabreu  @larroy - What is the next step here?

**Comment by larroy** [comment]
@marcoabreu @Chancebair @edisongustavo Do you want this PR? Want to take over? I don't have capacity to follow up on this one right now.

**Comment by marcoabreu** [comment]
The fix got merged in #13840, so I think we're fine here.

**Comment by larroy** [comment]
@marcoabreu you dismissed my refactor which greatly helps diagnose problems with this script.

**Comment by marcoabreu** [comment]
While I understand that environment variables will greatly speed up the duration of an iteration, I don't feel comfortable adding a hack to our configuration system. Right now, everything is defined as configuration in code that's entirely versioned. Introducing these varibales opens the possibilities to modify configuration without audit trail or approval. Considering that anybody can just change these values in a production system without notice or review, it doesn't seem like best practice to me to introduce this feature. 

If someone wants to iterate, I'd suggest to create a job in dev, create a branch and change the configuration values inside the code. But environment variables will have an impact on ALL jobs instead of being local to a specific branch or job.

**Comment by larroy** [comment]
@marcoabreu What hack are you talking about? Why should anyone reate a job in dev?

**Comment by marcoabreu** [comment]
No they were not: https://github.com/apache/incubator-mxnet/pull/13884/files#diff-76bb673252e787d0be4340b502e89719L37

**Comment by marcoabreu** [comment]
The only exception where environment variables were chosen was here: https://github.com/apache/incubator-mxnet/pull/13884/files#diff-76bb673252e787d0be4340b502e89719L193
This was necessary because these are deployment/environment-specific variables.

These variables on the other hand are being removed by you here:
https://github.com/apache/incubator-mxnet/pull/13884/files#diff-76bb673252e787d0be4340b502e89719R245


Thus, this change entirely inverts the whole existing design strategy for the configuration.

**Pull request closed without merging**